src/hooks/useStringMask.ts

```diff
@@ -0,0 +1,120 @@
+import { useMemo } from 'react';
+
+export type StringMaskToken = {
+    mask: string;
+    matchValue: (char: string) => boolean;
+};
+
+const maskTokens: StringMaskToken[] = [
+    {
+        mask: '0',
+        matchValue: (char: string) => char.match(/^0{1}$/gim) !== null,
+    },
+    {
+        mask: '1',
+        matchValue: (char: string) => char.match(/^[0-1]{1}$/gim) !== null,
+    },
+    {
+        mask: '2',
+        matchValue: (char: string) => char.match(/^[0-2]{1}$/gim) !== null,
+    },
+    {
+        mask: '3',
+        matchValue: (char: string) => char.match(/^[0-3]{1}$/gim) !== null,
+    },
+    {
+        mask: '4',
+        matchValue: (char: string) => char.match(/^[0-4]{1}$/gim) !== null,
+    },
+    {
+        mask: '5',
+        matchValue: (char: string) => char.match(/^[0-5]{1}$/gim) !== null,
+    },
+    {
+        mask: '6',
+        matchValue: (char: string) => char.match(/^[0-6]{1}$/gim) !== null,
+    },
+    {
+        mask: '7',
+        matchValue: (char: string) => char.match(/^[0-7]{1}$/gim) !== null,
+    },
+    {
+        mask: '8',
+        matchValue: (char: string) => char.match(/^[0-8]{1}$/gim) !== null,
+    },
+    {
+        mask: '9',
+        matchValue: (char: string) => char.match(/^[0-9]{1}$/gim) !== null,
+    },
+    {
+        mask: 'A',
+        matchValue: (char: string) => char.match(/^[A-Z]{1}$/gim) !== null,
+    },
+    {
+        mask: 'S',
+        matchValue: (char: string) => char.match(/^[A-Z0-9]{1}$/gim) !== null,
+    },
+];
+
+export const applyStringMask = (mask: string, value: string) => {
+    const maskChars = mask.split('');
+    let valueIndex = 0;
+    const output: string[] = [];
+
+    for (let i = 0; i < maskChars.length; i++) {
+        const maskChar = maskChars[i];
+        const token = maskTokens.find(t => t.mask === maskChar);
+
+        if (valueIndex >= value.length) {

```

Comment:
for loops in 2019? Can't this be done with map, reduce, forEach etc?

Reply:
Probably. I'll have a look at it. Bit of brain farts in there seems like

Reply:
@eskil0312 fixed